feat: Release `1.0.0-pre.9`

jamilbk opened the pull request:
This is a staging issue to update docs, website, and perform infrastructure changes as part of the `1.0.0-pre.9` release.

- [ ] https://firezonehq.slack.com/archives/C04HRQTFY0Z/p1707763144588859
- [ ] Explain that `Everyone` group is now automatic
- [ ] Migrate existing `sites.routing` data to use managed relays, explain to customers
- [ ] Manually test gateway upgrade script
- [x] Better client docs

jamilbk committed:
Checkpoint

jamilbk committed:
typos

AndrewDryga commented on website/src/app/kb/authenticate/entra/page.tsx:
```diff
@@ -0,0 +1,11 @@
+import { Metadata } from "next";
+
+export const metadata: Metadata = {
+  title: "Google Workspace Authentication • Firezone Docs",
```
```suggestion
  title: "Microsoft Entra Authentication • Firezone Docs",
```

AndrewDryga commented on website/src/app/kb/authenticate/entra/readme.mdx:
```diff
@@ -0,0 +1,262 @@
+The Firezone Microsoft Entra connector integrates with Microsoft's identity APIs
+to support user authentication and user / group sync.
+
+Users, groups, and organizational units are synced every few minutes to ensure
```
Let's not give `few minutes` promises, but rather say that we try to sync them at least once an hour or so. This will be better for the future when we have more issues with scaling workers.

jamilbk replied:
Yeah I don't think this conveys any sort of promise. What if I just add "every few minutes on average"? Synced once an hour sounds incredibly long

jamilbk replied:
Or maybe we can explain the architecture better in the User-group sync docs

AndrewDryga replied:
@jamilbk the problem is that we do full directory comparison and doing it every minute for 4k people org will be not efficient without webhooks (that are easy to add later). I would just be careful now saying it's every few minutes when it's every ~5-8 minutes right now and can increase later.

AndrewDryga commented on website/src/app/kb/authenticate/entra/readme.mdx:
```diff
@@ -0,0 +1,262 @@
+Click **CREATE** after you've filled in the fields above.
+
+<Image
+  src="/images/gcp-create-project.png"
```
This is GCP image

jamilbk replied:
Yeah PR is still a wip ;-)

AndrewDryga commented on website/src/app/kb/authenticate/user-group-sync/readme.mdx:
```diff
@@ -1,70 +1,73 @@
+  - john@company.com
 - Group:Engineering
   - bob@company.com
   - alice@company.com
```
We sync transient memberships:
```suggestion
  - alice@company.com
  - john@company.com
```

jamilbk committed:
Merge branch 'main' into docs/1.0.0-pre.9

jamilbk committed:
Beef up OIDC

AndrewDryga commented on website/src/app/docs/administer/upgrade/readme.mdx:
```diff
@@ -7,14 +7,11 @@ import Image from "next/image";
-`}
-/>
+<Alert color="info">
+  Automatic rollbacks are still under development. We recommend backing up
```
Id say we will never promise to ship them in new version

jamilbk replied:
(old docs)

jamilbk committed:
Merge branch 'main' into docs/1.0.0-pre.9

jamilbk committed:
Add Entra and Okta docs

jamilbk committed:
Fix elixir lint

jamilbk committed:
Merge branch 'main' into docs/1.0.0-pre.9

jamilbk committed:
Merge branch 'main' into docs/1.0.0-pre.9

jamilbk committed:
Wrap Image inside Link

jamilbk committed:
Checkpoint updates

jamilbk committed:
Merge branch 'main' into docs/1.0.0-pre.9

jamilbk committed:
Checkpoint

ReactorScram committed:
ci: document and fix a couple things for local Docker testing (#3672)

Signed-off-by: Reactor Scram <ReactorScram@users.noreply.github.com>
Co-authored-by: Jamil <jamilbk@users.noreply.github.com>

jamilbk committed:
feat(android): Add support for per-app VPN configurable through MDM (#3657)

Refs #3613

jamilbk committed:
chore(portal): Add okta as IdP in dev (#3675)

jasonboukheir committed:
fix(android): move log zip file outside of log folder (#3677)

Fixes #3330

The when the log zip file is created, it's created as an empty file in
the `logs` directory. The `logs` directory is then zipped, and written
to the zip file. The empty file remains as an artifact in the final zip
file.

This moves the location of the zip file outside of the `logs` directory,
where it will be cleaned up later.

jamilbk committed:
Checkpoint

jamilbk committed:
Checkpoint

AndrewDryga committed:
Revert "fix(gateway): Remove `/dev/net/tun` requirement and clean up upgrade script (#3691)

This reverts PR #3392.
This reverts commit 16f5401a73927a748ff5a91f808f8f2c94c9bd6a.

jamilbk committed:
checkpoint

jamilbk committed:
fix(ci): Ensure integration-tests allow for at least 30 seconds to establish a connection (#3676)

So the cause of the flaky tests is that they aren't waiting long enough
for a connection to be established. Both the test in #3666 and the
`iperf` tests have a timeout of 10 seconds.

Connections _should_ be established **very quickly** in CI. However, I
have a few guesses as to why they might not be, essentially causing us
to have to wait for a timeout to re-initiate a connection request:

- Packets arrive out of order or too quickly for the WireGuard state
machine to establish a handshake.
- Too many ICE candidates gathered (the gateway has 3 interfaces)


This PR:

- Refactors the iperf tests to be a little easier to maintain
- Ensures `integration-tests` run for at least 30 seconds before timing
out


In any case, we can debug / optimize this further after snownet is
merged, which might just solve the problem completely.

bmanifold committed:
feat(portal): Add sign-in success page for clients (#3659)

Why:

* On some clients, the web view that is opened to sign-in to Firezone is
left open and ends up getting stuck on the Sign In page with the
liveview loader on the top of the page also stuck and appearing as
though it is waiting for another response. This commit adds a sign-in
success page that is displayed upon successful sign-in and shows a
message to the user that lets them know they can close the window if
needed. If the client device is able to close the web view that was
opened, then the page will either very briefly be shown or will not be
visible at all due to how quickly the redirect happens.

Closes #3608 

<img width="625" alt="Screenshot 2024-02-15 at 4 30 57 PM"
src="https://github.com/firezone/firezone/assets/2646332/eb6a5df6-4a4c-4e54-b57c-5da239069ea9">

---------

Signed-off-by: Jamil <jamilbk@users.noreply.github.com>
Co-authored-by: Jamil <jamilbk@users.noreply.github.com>

jamilbk committed:
Revert "feat(portal): Add sign-in success page for clients" (#3692)

Merged a bit too soon!

jamilbk committed:
Checkpoint

dependabot[bot] committed:
build(deps): Bump semver from 1.0.21 to 1.0.22 in /rust (#3685)

Bumps [semver](https://github.com/dtolnay/semver) from 1.0.21 to 1.0.22.
<details>
<summary>Release notes</summary>
<p><em>Sourced from <a
href="https://github.com/dtolnay/semver/releases">semver's
releases</a>.</em></p>
<blockquote>
<h2>1.0.22</h2>
<ul>
<li>Fix unused_imports warnings when compiled by rustc 1.78</li>
</ul>
</blockquote>
</details>
<details>
<summary>Commits</summary>
<ul>
<li><a
href="https://github.com/dtolnay/semver/commit/c8ad1bf6db35b1847abbe8ddc7c8fba0455e5fc3"><code>c8ad1bf</code></a>
Release 1.0.22</li>
<li><a
href="https://github.com/dtolnay/semver/commit/f76db8d7f2f542ac45aa2d7e3c06afa428fe8698"><code>f76db8d</code></a>
Resolve redundant import warning</li>
<li><a
href="https://github.com/dtolnay/semver/commit/f32b420f75d060cdc451bfac091318ed8bc0e41f"><code>f32b420</code></a>
Ignore incompatible_msrv clippy lint for conditionally compiled
code</li>
<li>See full diff in <a
href="https://github.com/dtolnay/semver/compare/1.0.21...1.0.22">compare
view</a></li>
</ul>
</details>
<br />


[![Dependabot compatibility
score](https://dependabot-badges.githubapp.com/badges/compatibility_score?dependency-name=semver&package-manager=cargo&previous-version=1.0.21&new-version=1.0.22)](https://docs.github.com/en/github/managing-security-vulnerabilities/about-dependabot-security-updates#about-compatibility-scores)

Dependabot will resolve any conflicts with this PR as long as you don't
alter it yourself. You can also trigger a rebase manually by commenting
`@dependabot rebase`.

[//]: # (dependabot-automerge-start)
[//]: # (dependabot-automerge-end)

---

<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits
that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot squash and merge` will squash and merge this PR after
your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge
and block automerging
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot close` will close this PR and stop Dependabot recreating
it. You can achieve the same result by closing it manually
- `@dependabot show <dependency name> ignore conditions` will show all
of the ignore conditions of the specified dependency
- `@dependabot ignore <dependency name> major version` will close this
group update PR and stop Dependabot creating any more for the specific
dependency's major version (unless you unignore this specific
dependency's major version or upgrade to it yourself)
- `@dependabot ignore <dependency name> minor version` will close this
group update PR and stop Dependabot creating any more for the specific
dependency's minor version (unless you unignore this specific
dependency's minor version or upgrade to it yourself)
- `@dependabot ignore <dependency name>` will close this group update PR
and stop Dependabot creating any more for the specific dependency
(unless you unignore this specific dependency or upgrade to it yourself)
- `@dependabot unignore <dependency name>` will remove all of the ignore
conditions of the specified dependency
- `@dependabot unignore <dependency name> <ignore condition>` will
remove the ignore condition of the specified dependency and ignore
conditions


</details>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

jamilbk committed:
Update sync docs

jamilbk committed:
chore(repo): Update issue template with support links (#3678)

Fixes #3627

---------

Signed-off-by: Jamil <jamilbk@users.noreply.github.com>
Co-authored-by: Andrew Dryga <andrew@dryga.com>

conectado committed:
fix(snownet): don't trigger busy loop upon refreshing allocation (#3693)

Currently, after the refresh timeout the gateway or client starts
looping forever, since after the refresh timeout is reached the
`poll_timeout` will always return `ALLOCATION_LIFETIME/2`(which would be
"now"), since `allocation_lifetime` is never updated.

To fix this, we need to check whether we currently have a refresh
request in flight before queuing a new one. Additionally, we need to
abort refreshing as soon as the lifetime of the allocation expires.

Related: #3617.
Related: #3631.

---------

Co-authored-by: Thomas Eizinger <thomas@eizinger.io>

conectado committed:
refactor(connlib): replace `webrtc-rs` with `snownet` (#3391)

Co-authored-by: Thomas Eizinger <thomas@eizinger.io>

Resolves: #3377.

---------

Co-authored-by: Thomas Eizinger <thomas@eizinger.io>

AndrewDryga commented on elixir/apps/web/lib/web/live/settings/identity_providers/google_workspace/new.ex:
```diff
@@ -41,9 +41,10 @@ defmodule Web.Settings.IdentityProviders.GoogleWorkspace.New do
-        For a more detailed guide on setting up Firezone with Google Workspace, please <.link class={
-          link_style()
-        }>refer to our documentation</.link>.
+        For a more detailed guide on setting up Firezone with Google Workspace, please <.link
```
Should we do a `<.docs_ref path="authenticate/google">text</.docs_ref>` component?

jamilbk replied:
Good call will add

AndrewDryga commented on website/src/app/blog/how-dns-works-in-firezone/readme.mdx:
```diff
@@ -0,0 +1,59 @@
+
+When an application makes a DNS query, the proxy checks if the name matches a
+Firezone resource that the user has access to. If it does, the client then
+requests a gateway serving that resource to resolve the query. When the gateway
```
```suggestion
requests a gateway serving that resource to resolve the DNS query. When the gateway
```

AndrewDryga commented on website/src/app/blog/how-dns-works-in-firezone/readme.mdx:
```diff
@@ -0,0 +1,59 @@
+From this point onward, all subsequent packets to the dummy IP address are
+forwarded to the gateway that resolved the query.
+
+immediately responds with appropriate dummy IPs in the `100.64.96.0/11` and
```
This section seems off, probably a leftover from some copy-paste.

jamilbk replied:
Yeah, draft blog article

AndrewDryga commented on website/src/app/blog/how-dns-works-in-firezone/readme.mdx:
```diff
@@ -0,0 +1,59 @@
+responds, the proxy will create a temporary mapping between the actual IP
+address of the resource and the dummy IP address it returned to the application.
+From this point onward, all subsequent packets to the dummy IP address are
+forwarded to the gateway that resolved the query.
```
```suggestion
forwarded to the gateway that resolved the query. These IPs are persisted 
for the duration of the client's session.
```

AndrewDryga commented on website/src/app/blog/how-dns-works-in-firezone/readme.mdx:
```diff
@@ -0,0 +1,59 @@
+configured, the query is forwarded to the client device's default system
+resolver(s) instead.
+
+This means that clients are automatically configured for Split DNS in Firezone
```
Should this be the first part of the doc? Maybe for all long pages, we should start with TLDR like this one?

jamilbk replied:
This is a blog post, but yes each Docs page does start with an introductory sentence or two

AndrewDryga commented on website/src/app/docs/administer/regen-keys/readme.mdx:
```diff
@@ -11,24 +11,17 @@ do so using the same bootstrap scripts that were used when installing Firezone.
+</Alert>
+
+<Alert color="warning">
+  Replacing `GUARDIAN_SECRET_KEY`, `SECRET_KEY_BASE`, `LIVE_VIEW_SIGNING_SALT`,
```
So there are three things for the 1.0:
1. `DATABASE_ENCRYPTION_KEY` will render all data unusable, but we don't have it. And I would say unless we build it properly (with an ability to rotate the keys or use Vault for it) it's not really a good idea to build it in: it's a secret that you need to store along with your backups and it never expires, and if you lose it - you lost all the data.
2. Changing `TOKENS_KEY_BASE`, `TOKENS_SALT` will make all existing tokens unusable.
3. `SECRET_KEY_BASE`, `LIVE_VIEW_SIGNING_SALT`, `COOKIE_SIGNING_SALT`, `COOKIE_ENCRYPTION_SALT` will result in all sessions to be expired.

jamilbk replied:
Old docs, just being reformatted

AndrewDryga commented on website/src/app/kb/authenticate/google/readme.mdx:
```diff
@@ -1,27 +1,25 @@
+</PlanBadge>
+
+Firezone integrates with [Google Workspace](https://workspace.google.com) using
+a custom connector that supports both authenticating users and syncing users and
```
```suggestion
a direct API integration that supports both authenticating users and syncing users, org units and
```

jamilbk replied:
I call these `connectors` in the remainder of this page and other places of the docs, so going to leave it as-is.

jamilbk replied:
Maybe `API connector`

AndrewDryga commented on website/src/app/kb/authenticate/oidc/readme.mdx:
```diff
@@ -1,29 +1,115 @@
+  User / group sync is **not supported** with the universal OIDC connector. See
+  the [Google Workspace](/kb/authenticate/google), [Microsoft Entra
+  ID](/kb/authenticate/entra), or [Okta](/kb/authenticate/okta) connectors for
+  automatic user / group sync.
```
Just a nitpick: this way we are not repetitive and have an extra keyword for search indexers.
```suggestion
  automatic directory sync.
```

AndrewDryga commented on website/src/app/kb/authenticate/user-group-sync/readme.mdx:
```diff
@@ -1,70 +1,77 @@
-Deleting a group or organizational unit in Google Workspace will hide the group
-and associated policy in Firezone.
+Deleting a group or organizational unit in your identity provider will hide the
+group and disable the associated policy in Firezone.
```
```suggestion
group and delete the associated policy in Firezone.
```

AndrewDryga commented on website/src/app/kb/authenticate/user-group-sync/readme.mdx:
```diff
@@ -1,70 +1,77 @@
-Workspace. User membership is determined **only** by its immediate parent. For
-example, if you had the following group structure in your Google Workspace
-account:
+Firezone syncs transitive memberships from your identity provider. This means
```
@bmanifold just wanted to confirm if we do it for Okta too? Probably it should be a pre-provider feature description and statement that we try to do it whenever the provider supports fetching transitive memberships.

AndrewDryga commented on website/src/app/kb/deploy/dns/readme.mdx:
```diff
@@ -0,0 +1,93 @@
+making the query.
+
+DNS responses generated by the proxy have a TTL of **5 minutes** to ensure that
+clients will occasionally re-query the proxy to find the best gateway to serve a
```
It will not trigger a gateway reconnection, DNS TTL has nothing to do with our traffic routing, it's just for cases when upstream DNS changes IPs (and we must actually read TTL and use it in the future, and set 3 sec TTL on clients for resource domains).

jamilbk replied:
Hm this raises an interesting point.

@conectado How does failover work for DNS resources? If I've created a DNS mapping to a gateway, and the gateway goes down, will the mapping automatically recover? Or will the application need to request a new IP for the resource?

jamilbk replied:
If the application needs to request a new IP for the resource, it will only do so after 5 minutes, correct? And many applications simply won't.

Refs #3369

conectado replied:
The TTL of the local response is set to 1second, every 5 minutes we refresh the dns mapping, but if the connection to the gateway stops the mapping is invalidated.

Most stateful connections will need to be re-established if the connection fails and will therefore re-resolve the address. Save for ping there aren't many applications that after connections stop and start again don't resolve the address.

The mapping can't be automatically recovered, though we could try to do a best effort, because the mapping is between IPs and not DNS names.

conectado replied:
> It will not trigger a gateway reconnection, DNS TTL has nothing to do with our traffic routing, it's just for cases when upstream DNS changes IPs (and we must actually read TTL and use it in the future, and set 3 sec TTL on clients for resource domains).

Yes, this, it's an open issue #3010 we should refresh the dns mappings every TTL seconds instead of 5 minutes but this is independent from the TTL we set for the record

jamilbk replied:
> Save for ping there aren't many applications that after connections stop and start again don't resolve the address.

I think this was my question: it's the DNS query itself that triggers the resource_intent correct? So as long as as the connection is healthy, a new gateway will never be picked.

If the connection to the gateway times out, the application will need to requery DNS again, which typically (we're assuming) will happen when it tries to re-establish its connectivity.

Do I have that right?

conectado replied:
> I think this was my question: it's the DNS query itself that triggers the resource_intent correct? So as long as as the connection is healthy, a new gateway will never be picked.

> If the connection to the gateway times out, the application will need to requery DNS again, which typically (we're assuming) will happen when it tries to re-establish its connectivity.

> Do I have that right?

Precisely!

AndrewDryga commented on website/src/app/kb/deploy/gateways/readme.mdx:
```diff
@@ -31,31 +37,60 @@ you'll need to make sure the following outbound traffic is allowed:
+number of overall gateways you need to deploy, possibly making it more difficult
+to manage and monitor them. Firezone's control plane is designed to make this as
+easy as possible, but it's still something to keep in mind when planning your
+deployment.
```
Maybe we should push more for zero trust here rather than convincing that it "will be hard"? I would even reference a few standards/good sources explaining why zero trust is important.

jamilbk replied:
Updated to say

> The downside of deploying gateways close to resources is that it increases the
number of overall gateways you need to manage and monitor. Firezone was
carefully designed however to minimize this downside, making it easy to manage
even hundreds of gateways from the admin portal.

AndrewDryga commented on website/src/app/kb/deploy/gateways/readme.mdx:
```diff
@@ -31,31 +37,60 @@ you'll need to make sure the following outbound traffic is allowed:
+| ------------ | ------------ | --------- | ------ | ------------ |
+| Small        | 100          | 2         | 2 GB   | 1 Gbps       |
+| Medium       | 1,000        | 8         | 8 GB   | 10 Gbps      |
+| Large        | 10,000       | 32        | 32 GB  | 40 Gbps      |
```
We should test first, IRL I think we are CPU and network IO bound and RAM should not matter that much.

jamilbk replied:
I've done some testing on AWS. With plain WireGuard, we're CPU and network I/O bound.

The gateway does use a non-trivial amount of memory, so these sizing recommendations are just to get started.

AndrewDryga commented on website/src/app/kb/deploy/gateways/readme.mdx:
```diff
@@ -31,31 +37,60 @@ you'll need to make sure the following outbound traffic is allowed:
+
+You can deploy multiple gateways from the admin portal, or automate the process
+by reusing the `FIREZONE_TOKEN` environment variable shown when deploying a
+gateway from the admin portal.
```
```suggestion
gateway from the admin portal. All gateways within the sate **must** operate in the same network environment.
```

jamilbk replied:
This is mentioned more explicitly in the Sites documentation and at the top of the deploy docs

AndrewDryga commented on website/src/app/kb/deploy/gateways/readme.mdx:
```diff
@@ -64,32 +99,56 @@ This means you can automate deployment of the gateway using your automation
+  gateways they're connected to. If the keepalive timer expires, the client will
+  disconnect from that gateway and ask the portal for a new, healthy one to
+  connect to.
+- On average, the client keepalive timer expires after **20 seconds**. This is
```
Hope we will be able to tune it down to 2-5 in the future, being on a video call on the go is not a joke anymore with Vision Pro ppl :).

AndrewDryga commented on website/src/app/kb/deploy/sites/readme.mdx:
```diff
@@ -1,17 +1,33 @@
+
+A site can be as small as a single gateway running on a single server managing
+the access to the resources on that server, or as large as an entire data center
+managing access to an entire fleet of servers.
```
I think the data center will not have a network environment where everything can access everything, maybe we should align the site explanation closer to VPC or some DevOps term well understood by our users?

jamilbk replied:
Sure, a better description here would be "data center rack"

jamilbk replied:
Updated to say "data center network managing access to an entire cluster of servers". This will be common for AI workloads.

AndrewDryga commented on website/src/app/kb/quickstart/readme.mdx:
```diff
@@ -80,8 +93,9 @@ Site Details page where the gateway will now be listed and shown as online.
-access to. Examples include servers, databases, subnets, etc...
+A Resource is simply anything the gateway can reach that you'd like to give
+users access to. Examples include servers, databases, applications, and even
+entire subnets.
```
I would suggest something like this: 
```suggestion
entire subnets.

The more granular resources are the better your control over the security policies around it.
```
CIDR ranges will be common but should not be encouraged.

jamilbk replied:
I'll this inside an `Alert color=info`

AndrewDryga committed:
feat(portal): Billing system (#3642)

jamilbk committed:
fix(ci): Run each perf test in its own matrix job (#3695)

The iperf3 server sometimes hangs, or takes a while to startup.

Rather than trying to reset the iperf3 state between performance tests,
this PR refactors them so they each run in their matrix job. This
ensures each performance test will run on a separate VM, unaffected by
previous test runs to eliminate the effect any residual network buffer
state can have on a particular test.

It also makes sure the server is listening with a `healthcheck`.

jamilbk committed:
fix(ci): Fix result overwrite (#3700)

Buttoning up fixes from #3695

jamilbk committed:
fix(ci): Fix typo preventing perf-results from being calculated (#3703)

dependabot[bot] committed:
build(deps): Bump ring from 0.17.7 to 0.17.8 in /rust (#3686)

Bumps [ring](https://github.com/briansmith/ring) from 0.17.7 to 0.17.8.
<details>
<summary>Commits</summary>
<ul>
<li>See full diff in <a
href="https://github.com/briansmith/ring/commits">compare view</a></li>
</ul>
</details>
<br />


[![Dependabot compatibility
score](https://dependabot-badges.githubapp.com/badges/compatibility_score?dependency-name=ring&package-manager=cargo&previous-version=0.17.7&new-version=0.17.8)](https://docs.github.com/en/github/managing-security-vulnerabilities/about-dependabot-security-updates#about-compatibility-scores)

Dependabot will resolve any conflicts with this PR as long as you don't
alter it yourself. You can also trigger a rebase manually by commenting
`@dependabot rebase`.

[//]: # (dependabot-automerge-start)
[//]: # (dependabot-automerge-end)

---

<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits
that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot squash and merge` will squash and merge this PR after
your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge
and block automerging
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot close` will close this PR and stop Dependabot recreating
it. You can achieve the same result by closing it manually
- `@dependabot show <dependency name> ignore conditions` will show all
of the ignore conditions of the specified dependency
- `@dependabot ignore <dependency name> major version` will close this
group update PR and stop Dependabot creating any more for the specific
dependency's major version (unless you unignore this specific
dependency's major version or upgrade to it yourself)
- `@dependabot ignore <dependency name> minor version` will close this
group update PR and stop Dependabot creating any more for the specific
dependency's minor version (unless you unignore this specific
dependency's minor version or upgrade to it yourself)
- `@dependabot ignore <dependency name>` will close this group update PR
and stop Dependabot creating any more for the specific dependency
(unless you unignore this specific dependency or upgrade to it yourself)
- `@dependabot unignore <dependency name>` will remove all of the ignore
conditions of the specified dependency
- `@dependabot unignore <dependency name> <ignore condition>` will
remove the ignore condition of the specified dependency and ignore
conditions


</details>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

thomaseizinger committed:
fix(connlib): set socket to IPv6 only before binding address (#3701)

Binding the IPv6 socket would always fail because we do it _after_ the
IPv4 socket. However, without setting the socket to `IP6_ONLY`, binding
the unspecified IPv6 address (`::`) also wants to bind to IPv4
(depending on the kernel settings). To avoid this, we need to first
configure the socket properly and then bind it to the given address.

Co-authored-by: Jamil <jamilbk@users.noreply.github.com>

jamilbk committed:
fix(ci): Fix js typo (#3704)

More fixes from the perf test refactor

jamilbk committed:
chore(ci): enforce shellscript formatting and style (#3679)

Noticed that we all have different styles of writing scripts :-).

This PR adds linting to our shell scripts to standardize on formatting,
catch common issues and/or possible security bugs.

For editor setup:
- Ensure [`shellcheck`](https://github.com/koalaman/shellcheck) and
[`shfmt`](https://github.com/mvdan/sh) are in your `PATH`
- Configure `shfmt` with indentation of `4`, otherwise it uses tabs by
default.
[Here](https://github.com/jamilbk/nvim/blob/master/init.vim#L159) is how
you can do that with Vim and
[here](https://marketplace.visualstudio.com/items?itemName=mkhl.shfmt)
is how for VScode.

---------

Signed-off-by: Jamil <jamilbk@users.noreply.github.com>
Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: Reactor Scram <ReactorScram@users.noreply.github.com>
Co-authored-by: Thomas Eizinger <thomas@eizinger.io>
Co-authored-by: Brian Manifold <bmanifold@users.noreply.github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Andrew Dryga <andrew@dryga.com>
Co-authored-by: Gabi <gabrielalejandro7@gmail.com>

thomaseizinger committed:
feat!(portal): return relays as plain socket addresses (#3665)

Extracted out of #3391.

We don't actually need this for #3391 though because we've added a
compatibility layer during deserialization. But, it will be good to
remove that compat layer at some point which means we have to return the
addresses as plain socket addresses. Because that is a breaking change,
I decided to extract this into a different PR.

Co-authored-by: conectado <gabrielalejandro7@gmail.com>

---------

Co-authored-by: conectado <gabrielalejandro7@gmail.com>

conectado committed:
refactor(connlib): simplify resource ordering by implementing Ord trait (#3696)

This is a small refactor that could be extracted from the work I'm doing
for #2022

Co-authored-by: Jamil <jamilbk@users.noreply.github.com>

jamilbk committed:
refactor(ci): Merge perf results into one comment (#3707)

One comment vs eight, need I say more?

thomaseizinger committed:
feat(connlib): filter out relays based on our locally created sockets (#3705)

Currently, we will always try to reach all relays that we are given by
the portal. That creates unnecessary warnings if we don't have
connectivity for a certain IP version.

By filtering based on the IP version of our bound sockets, we can avoid
these warnings in the logs.

---------

Co-authored-by: Jamil <jamilbk@users.noreply.github.com>

dependabot[bot] committed:
build(deps): Bump arboard from 3.3.0 to 3.3.1 in /rust (#3687)

Bumps [arboard](https://github.com/1Password/arboard) from 3.3.0 to
3.3.1.
<details>
<summary>Release notes</summary>
<p><em>Sourced from <a
href="https://github.com/1Password/arboard/releases">arboard's
releases</a>.</em></p>
<blockquote>
<h2>v3.3.1</h2>
<h3>Changed</h3>
<ul>
<li>Updated Windows clipboard and migrated from <code>winapi</code> to
<code>windows-sys</code>.</li>
<li>Internally migrated to Rust 2021 edition.</li>
<li>Significantly improved the crate's error documentation.</li>
<li>Updated <code>core-graphics</code> to <code>0.23</code></li>
<li>Updated <code>x11rb</code> to <code>0.13</code></li>
</ul>
<p><strong>Full Changelog</strong>: <a
href="https://github.com/1Password/arboard/compare/v3.3.0...v3.3.1">https://github.com/1Password/arboard/compare/v3.3.0...v3.3.1</a></p>
</blockquote>
</details>
<details>
<summary>Changelog</summary>
<p><em>Sourced from <a
href="https://github.com/1Password/arboard/blob/master/CHANGELOG.md">arboard's
changelog</a>.</em></p>
<blockquote>
<h2>3.3.1 on 2024-12-02</h2>
<h3>Changed</h3>
<ul>
<li>Updated Windows clipboard and migrated from <code>winapi</code> to
<code>windows-sys</code>.</li>
<li>Internally migrated to Rust 2021 edition.</li>
<li>Significantly improved the crate's error documentation.</li>
<li>Updated <code>core-graphics</code> to <code>0.23</code></li>
<li>Updated <code>x11rb</code> to <code>0.13</code></li>
</ul>
</blockquote>
</details>
<details>
<summary>Commits</summary>
<ul>
<li><a
href="https://github.com/1Password/arboard/commit/77e0e078eb460ac2fa0eda96124163c11ef6b2d1"><code>77e0e07</code></a>
Release 3.3.1</li>
<li><a
href="https://github.com/1Password/arboard/commit/409bd98978940b1a14a2280bea4b68800f353668"><code>409bd98</code></a>
Update x11rb to 0.13 and core-graphics to 0.23</li>
<li><a
href="https://github.com/1Password/arboard/commit/bd91f9c438294cf894dce45051fb48c0759705dc"><code>bd91f9c</code></a>
Increase error documentation on Clipboard type</li>
<li><a
href="https://github.com/1Password/arboard/commit/a648570ce9c8d7051546f93df4b06cd6e82e9e01"><code>a648570</code></a>
Update CI actions</li>
<li><a
href="https://github.com/1Password/arboard/commit/0d6725d97f91064e90a7c238ac6a311c0a06ee29"><code>0d6725d</code></a>
Spell check docs</li>
<li><a
href="https://github.com/1Password/arboard/commit/a100f2d77c5b2877257220e9144179d1b78784fe"><code>a100f2d</code></a>
Update clipboard-win to v5 and replace winapi with windows-sys (<a
href="https://redirect.github.com/1Password/arboard/issues/123">#123</a>)</li>
<li><a
href="https://github.com/1Password/arboard/commit/1b8df75ee234dc8ae13785528bd34e4cb1aba8c3"><code>1b8df75</code></a>
Bump to Rust 2021 edition</li>
<li><a
href="https://github.com/1Password/arboard/commit/e3f54c30498f0bc3d7d716a1353e2f14a31bfa73"><code>e3f54c3</code></a>
Document MSRV of 1.61</li>
<li><a
href="https://github.com/1Password/arboard/commit/8c475cfd1420fe63f9fe21e62b4bf3d0b42f4175"><code>8c475cf</code></a>
Make winapi crate optional</li>
<li>See full diff in <a
href="https://github.com/1Password/arboard/compare/v3.3.0...v3.3.1">compare
view</a></li>
</ul>
</details>
<br />


[![Dependabot compatibility
score](https://dependabot-badges.githubapp.com/badges/compatibility_score?dependency-name=arboard&package-manager=cargo&previous-version=3.3.0&new-version=3.3.1)](https://docs.github.com/en/github/managing-security-vulnerabilities/about-dependabot-security-updates#about-compatibility-scores)

Dependabot will resolve any conflicts with this PR as long as you don't
alter it yourself. You can also trigger a rebase manually by commenting
`@dependabot rebase`.

[//]: # (dependabot-automerge-start)
[//]: # (dependabot-automerge-end)

---

<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits
that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot squash and merge` will squash and merge this PR after
your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge
and block automerging
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot close` will close this PR and stop Dependabot recreating
it. You can achieve the same result by closing it manually
- `@dependabot show <dependency name> ignore conditions` will show all
of the ignore conditions of the specified dependency
- `@dependabot ignore <dependency name> major version` will close this
group update PR and stop Dependabot creating any more for the specific
dependency's major version (unless you unignore this specific
dependency's major version or upgrade to it yourself)
- `@dependabot ignore <dependency name> minor version` will close this
group update PR and stop Dependabot creating any more for the specific
dependency's minor version (unless you unignore this specific
dependency's minor version or upgrade to it yourself)
- `@dependabot ignore <dependency name>` will close this group update PR
and stop Dependabot creating any more for the specific dependency
(unless you unignore this specific dependency or upgrade to it yourself)
- `@dependabot unignore <dependency name>` will remove all of the ignore
conditions of the specified dependency
- `@dependabot unignore <dependency name> <ignore condition>` will
remove the ignore condition of the specified dependency and ignore
conditions


</details>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Jamil <jamilbk@users.noreply.github.com>

thomaseizinger committed:
fix(snownet): make TURN channel bindings more reliable (#3708)

Previously, we would only bind channels for _established_ connections.
This caused a problem if we'd get the other parties candidates before
the offer response. Additionally, we'd often send multiple channel
bindings for the same peer which caused additional warnings in the logs.

jamilbk committed:
fix(ci): Use workflow id to fetch perf results (#3710)

jamilbk committed:
Update website/src/app/blog/how-dns-works-in-firezone/readme.mdx

Co-authored-by: Andrew Dryga <andrew@dryga.com>
Signed-off-by: Jamil <jamilbk@users.noreply.github.com>

jamilbk committed:
Update website/src/app/blog/how-dns-works-in-firezone/readme.mdx

Co-authored-by: Andrew Dryga <andrew@dryga.com>
Signed-off-by: Jamil <jamilbk@users.noreply.github.com>

jamilbk committed:
fix artifact from merge

jamilbk committed:
Merge remote-tracking branch 'origin/main' into docs/1.0.0-pre.9

jamilbk committed:
Update website/src/app/kb/authenticate/oidc/readme.mdx

Co-authored-by: Andrew Dryga <andrew@dryga.com>
Signed-off-by: Jamil <jamilbk@users.noreply.github.com>

jamilbk committed:
Update website/src/app/kb/authenticate/user-group-sync/readme.mdx

Co-authored-by: Andrew Dryga <andrew@dryga.com>
Signed-off-by: Jamil <jamilbk@users.noreply.github.com>

jamilbk committed:
Merge branch 'main' into docs/1.0.0-pre.9

jamilbk committed:
More updates

jamilbk committed:
Make formatting consisitent and use button for nextstep

jamilbk committed:
Fix typos

jamilbk committed:
Deploying gateways

jamilbk committed:
Finished

jamilbk committed:
Merge branch 'main' into docs/1.0.0-pre.9

ReactorScram commented on website/src/app/docs/deploy/advanced/build-from-source/readme.mdx:
```diff
@@ -4,43 +4,34 @@ import Alert from "@/components/DocsAlert";
+  [here](https://github.com/firezone/firezone/blob/legacy/.tool-versions) for
+  the versions we use for Erlang, Elixir, and Node. If your system supports it,
+  you can install these using
+  [asdf-vm](https://asdf-vm.com/guide/getting-started.html) using a similar
```
Might have to de-dupe this with https://github.com/firezone/firezone/blob/main/docs/CONTRIBUTING.md#asdf-vm-setup some time

jamilbk replied:
Yeah, probably need to update this section quite substantially.

These docs are more for those looking to build from source to deploy, and not necessarily to develop on

ReactorScram commented on website/src/app/docs/deploy/advanced/build-from-source/readme.mdx:
```diff
@@ -95,12 +86,9 @@ up. Once that's done, you can use `firezone start` to start Firezone and run
+<Alert color="info">
+  As mentioned at the top, it's recommended to use some sort of service
+  management to start and stop Firezone easily without having to manually do it
+  using the `firezone` binary directly. But the choice is yours, since you're in
```
For gateways, right? We host the portal and relay, and the Linux client isn't out, so systemd and "firezone binary" means a self-hosted gateway?

jamilbk replied:
Legacy docs

ReactorScram commented on website/src/app/docs/deploy/advanced/external-database/readme.mdx:
```diff
@@ -17,15 +17,12 @@ database services like Amazon RDS. See the
-`}
-/>
+<Alert color="warning">
+  Configuring Firezone to use an external database can be complicated and
```
Is this Firezone the gateway or Firezone the portal? The gateway doesn't need a DB, right?

jamilbk replied:
Correct, but this is for Firezone 0.7. These docs are around for legacy users and I didn't want to move them because SEO. If you view them, https://docs.firezone.dev, you'll see an alert at the top.

Maybe that needs to be more prominent?

ReactorScram replied:
It's all right, it could be a little taller. I missed it since I was just looking at the diff

ReactorScram commented on website/src/app/docs/deploy/advanced/reverse-proxy/readme.mdx:
```diff
@@ -2,16 +2,13 @@ import Alert from "@/components/DocsAlert";
-is suitable for the vast majority of use cases and is recommended for most
-users. There are important security risks if the reverse proxy is not set up
-correctly.
-`}
-/>
+<Alert color="warning">
+  Using a custom reverse proxy is an advanced configuration. The default bundled
+  Nginx proxy (Omnibus-based deployments) and Caddy (Docker-based deployments)
+  is suitable for the vast majority of use cases and is recommended for most
+  users. There are important security risks if the reverse proxy is not set up
+  correctly.
+</Alert>
```
Same question, this is for the portal / web UI, right? Since gateways and relays and clients don't have a web UI.

"Portal", "API", "control plane", "web app", and "web UI" are all terms for the same thing, right? And that's what's called "Firezone" in the docs?

jamilbk replied:
Well -- these are the legacy docs (notice `/docs` in the path) -- but yes, mostly. These are the synonymous groupings:

- `api`, `control plane`
- `portal`, `web app`, `web ui`

ReactorScram commented on website/src/app/docs/deploy/docker/readme.mdx:
```diff
@@ -158,13 +155,10 @@ on start.
-`}
-/>
+<Alert color="info">
+  Firezone currently uses WireGuard's [open-source client
+  apps](https://www.wireguard.com/install).
```
Huh, didn't know that.

jamilbk replied:
Outdated docs ;-). I probably should have moved this refactoring of the `Alert` component to a separate PR

ReactorScram commented on website/src/app/docs/deploy/docker/readme.mdx:
```diff
@@ -32,14 +32,11 @@ Docker offers a number of benefits over the old
-/>
+<Alert color="warning">
+  Before deploying Firezone in **production**, you'll need a valid DNS record
+  pointing to this instance. See [Prepare to
```
This instance of the portal or of some other component(s)? Further up this document it says "As of 0.6.0". That also applies to 0.7 and 1.0 then, right?

It also says ports 80, 443, and "`51820/udp`: VPN traffic listen port." I didn't know the portal handles tunneled traffic? I thought the web sockets ran over tcp/443 and then the portal just told the clients, relays, and gateways how to find each other? Does the portal act as a STUN server? Or does the Docker setup here include both the portal and relay components?

tbh I feel more confused than I think I should be after 3 months on the project lol

ReactorScram commented on website/src/app/docs/deploy/omnibus/readme.mdx:
```diff
@@ -4,18 +4,14 @@ import SupportOptions from "@/components/SupportOptions";
-/>
+<Alert color="warning">
+  Due to Omnibus being EOL'd by Chef in 2024, Docker is now the preferred method
+  of deploying Firezone. See the [Docker deployment guide](/docs/deploy/docker)
```
Same question

ReactorScram commented on website/src/app/docs/deploy/omnibus/readme.mdx:
```diff
@@ -30,24 +26,18 @@ to get started.
+</Alert>
+
+<Alert color="info">
+  Firezone modifies the kernel netfilter and routing tables. Other programs that
```
Surely it doesn't modify them on the portal? This must be talking about the relays?

ReactorScram commented on website/src/app/docs/deploy/readme.mdx:
```diff
@@ -94,13 +94,10 @@ You have two options for deploying Firezone:
+<Alert color="warning">
+  Chef Infra Client, the configuration system Chef Omnibus relies on, has been
+  [scheduled for End-of-Life in 2024](https://docs.chef.io/versions). As such,
+  support for Omnibus-based deployments will be removed starting with Firezone
+  0.8. To transition to Docker from Omnibus today, follow our [migration
```
didn't know there was a 0.8

jamilbk replied:
Yeah I should really comb through the old docs and finalize / seal them.

ReactorScram commented on website/src/app/docs/user-guides/client-instructions/readme.mdx:
```diff
@@ -102,16 +102,11 @@ your network admin. If you are already logged into the portal, click the
-`}
-/>
+<Alert color="info">
+  Importing the configuration file using the Network Manager GUI may fail with
+  the following error if the profile has IPv6 support enabled: `ipv6.method:
+  method "auto" is not supported for WireGuard`
```
Just saying, I didn't test any of the Network Manager stuff, so I can't confirm this

ReactorScram commented on website/src/app/docs/user-guides/client-instructions/readme.mdx:
```diff
@@ -164,16 +159,11 @@ Using `nmcli`, import the downloaded configuration file:
-`}
-/>
+<Alert color="info">
+  The WireGuard connection/interface will match the name of the configuration
```
Wireguard?

ReactorScram commented on website/src/app/docs/user-guides/use-cases/split-tunnel/readme.mdx:
```diff
@@ -50,13 +47,10 @@ the `/settings/default` page to override this value.
-  color="info"
-  html={`
+<Alert color="info">
   If you wish to resolve DNS queries over the encrypted VPN tunnel
```
Is the doctrine still "Don't call it VPN"?

jamilbk replied:
Yes, this is old stuff

ReactorScram commented on website/src/app/kb/authenticate/readme.mdx:
```diff
@@ -1,24 +1,45 @@
-- Service account: **365 days** (by default, configurable per token)
+| Component           | Auth Provider                     | Lifetime                                                                    |
+| ------------------- | --------------------------------- | --------------------------------------------------------------------------- |
+| Admin portal web UI | Email authentication              | **10 hours**                                                                |
```
Is there a non-admin portion of the web UI, or is "Admin portal" the same as "portal" and "Web UI" in general?

ReactorScram commented on website/src/app/kb/authenticate/readme.mdx:
```diff
@@ -1,24 +1,45 @@
 
-When a client application or service account token expires, the client is
+When a session token expires or is revoked, the affected component is
 disconnected immediately and must reauthenticate to regain access to resources.
```
And in the case of a client connecting directly to a gateway via LAN or STUN, this is enforced by the portal telling the client and gateway "This is how long the other computer's token lasts, disconnect them if I don't refresh them before that, and if I say revoke, disconnect them instantly"?

jamilbk replied:
Exactly. The portal will disconnect the client's websocket. The client will then try to reconnect it, receive a `401`, and then call `onDisconnect(error=401Unauthorized)`

jamilbk replied:
If someone is using a custom client that misbehaves by swallowing the `401`, the client connection to the gateway will linger a bit longer until the connection  needs to be refreshed.

@conectado Do you know how long that will take on average and worst-case?

ReactorScram replied:
Yeah I was thinking an official gateway would eventually force-disconnect a tampered client, and an official client would eventually force-disconnect a tampered gateway. 

I guess if it's in the case of an employee leaving a company, there's a chance they would have a tampered client, but tampered gateways are not very likely, and it's impossible to do anything anyway if they're both tampered.

ReactorScram commented on website/src/app/kb/deploy/dns/readme.mdx:
```diff
@@ -0,0 +1,93 @@
+- IPv6: `fd00:2021:1111:8000::100:100:111:0/120`
+
+When a user signs in, the client configures the host operating system to use
+this proxy as the default resolver for all queries on the system.
```
This is all intentionally identical to [the blog entry about DNS](https://github.com/firezone/firezone/blob/b6410755e06160a7558dd575b91113e37e92b84f/website/src/app/blog/how-dns-works-in-firezone/readme.mdx), right? Should the blog link here in case the docs change and the blog is kept as a historical artifact?

jamilbk replied:
Yeah that's a good idea. I plan for the blog post to be insanely detailed and read like a story, while the docs serve as a summarized reference.

Lots of technical content to write...

ReactorScram commented on website/src/app/kb/deploy/gateways/readme.mdx:
```diff
@@ -31,31 +37,67 @@ you'll need to make sure the following outbound traffic is allowed:
+
+<Alert color="warning">
+  When deploying gateways, remember that all gateways and resources in a site
+  must have unobstructed network connectivity to each other. This is necessary
```
Oh, so if I have 2 systems both running a gateway and a resource, I can't set up the resource to only accept localhost connections. e.g. If I run a web server behind Nginx, my web server only needs to accept from localhost, but if Nginx is in turn behind Firezone, Nginx still needs to accept from any gateway in my Firezone site.

And if that's a big problem, I would just split up the site into two smaller sites with  1 resource and 1 gateway each.

jamilbk replied:
Correct. We debated the naming of "Sites" for a long time. Other possible names:

- Networks
- Gateway Groups
- Subnets
- Clusters
- Contexts

In the end, "Site" seemed to most closely match users' expectation for how these things are run.

ReactorScram commented on website/src/app/kb/deploy/gateways/readme.mdx:
```diff
@@ -31,31 +37,67 @@ you'll need to make sure the following outbound traffic is allowed:
-are thus resource efficient by nature. A single gateway running on a 2 vCPU VM
-with 1 GB of memory should be able to handle hundreds of client connections
-under typical usage patterns.
+are thus resource efficient by nature.
```
🦀

ReactorScram commented on website/src/app/kb/deploy/gateways/readme.mdx:
```diff
@@ -31,31 +37,67 @@ you'll need to make sure the following outbound traffic is allowed:
-speeds can be achieved with syscall optimization and multi-threaded
-optimizations, both of which are currently being worked on.
+A single client connection to a single gateway can typically reach speeds of **1
+Gbps** or more. This scales linearly with each client connection up to the
```
Wow, on all the non-Windows clients?

jamilbk replied:
In my testing with boringtun on my 10gbe home network, yes, I hit about 1.1 Gbps once the connection was established.

This is between an M1 Macbook Air and a Ryzen 5 Linux desktop running the gateway.

ReactorScram commented on website/src/app/kb/deploy/gateways/readme.mdx:
```diff
@@ -64,32 +106,56 @@ This means you can automate deployment of the gateway using your automation
+<Alert color="warning">
+  **Note:** Be sure to set a unique `FIREZONE_ID` for each gateway you deploy.
+  This can be any non-empty string and is used to identify the gateway in the
+  portal for audit trail and logging purposes.
```
For some reason I thought `FIREZONE_ID` was the account slug. Is it the same for the clients, then? I'd been passing `--firezone-id firezone` to the Linux client, oops

To make it fool-proof (I know it's too late to change it but just saying) I would call it `FIREZONE_DEVICE_ID`

ReactorScram commented on website/src/app/kb/deploy/sites/readme.mdx:
```diff
@@ -1,17 +1,39 @@
+organization.
+
+A site can be as small as a single gateway running on a single server managing
+the access to the resources on that server, or as large as an entire data center
```
This answers my question from the other comment.

ReactorScram commented on website/src/app/kb/deploy/users/readme.mdx:
```diff
@@ -0,0 +1,38 @@
+Users with the `User` role can only sign in to the Firezone clients to access
+protected resources, while users with the `Admin` can sign in to the admin
+portal to perform administrative tasks as well as sign in to Firezone clients to
+access protected resources.
```
This answers my question from earlier. So "admin portal" == "portal" == "web UI", there is no non-admin portal.

ReactorScram commented on website/src/app/kb/quickstart/readme.mdx:
```diff
@@ -80,26 +93,36 @@ Site Details page where the gateway will now be listed and shown as online.
 
 - IP address
 - Fully Qualified Domain Name (FQDN)
 - CIDR Range
```
```suggestion
- IP address (e.g. 10.1.2.3)
- Fully Qualified Domain Name (FQDN) (e.g. http://gitlab.company.com)
- CIDR Range (e.g. 10.1.2.0/24)
```

I always just skim for examples before reading prose in docs

ReactorScram commented on website/src/app/kb/readme.mdx:
```diff
@@ -6,17 +6,29 @@ import Image from "next/image";
+## What is Firezone?
+
+[Firezone](https://www.firezone.dev) is an open source platform to securely
+manage remote access for any-sized organization. Unlike most VPNs, Firezone
```
"Unlike most VPNs" Yeah I would really like to just call it a VPN because people ask what my job is and I say, "It's not a VPN but, wink, it's a VPN"

jamilbk replied:
If those people aren't customers, call it a VPN ;-)

ReactorScram commented on website/src/app/kb/reference/glossary/readme.mdx:
```diff
@@ -0,0 +1,33 @@
+# Glossary of terms
+
```
Alphabetize, also:

```suggestion

**Account slug**: Your organization's tenant name, unique across all Firezone customers, e.g. `international-widget-corp-inc`

**Firezone ID**: AKA "Device ID", a non-empty string that identifies Clients and Gateways, unique within your organization, e.g. `johns-laptop` or `01HQP2K1AM41XWWNBVZS39N062`

**Admin portal**: AKA "portal" or "Web app", the central control plane where admin users create, manage, and delete Resources, Gateways, and Service Tokens for headless clients.
```

Hope the noob perspective helps lol

jamilbk replied:
I'll save `FIREZONE_ID` for an upcoming env var reference I'm planning to write.

ReactorScram commented on website/src/app/kb/reference/glossary/readme.mdx:
```diff
@@ -0,0 +1,33 @@
+**Resource**: A [Resource](/kb/deploy/resources) is any DNS name, IP, or network
+(CIDR range) you wish to manage access for. DNS-based Resources can be used to
+manage access to internal or external applications and automatically match all
+subdomains as well. CIDR-based Resources can be used to manage access for an
```
(also commented on the Resources readme) Wait if they automatically match subdomains, do I need `*.github.com` or `?.github.com` or just `github.com`?

jamilbk replied:
Ah, you've found a bug

ReactorScram commented on website/src/app/kb/deploy/resources/readme.mdx:
```diff
@@ -10,11 +17,12 @@ are expected to be reachable by all gateways in the same site.
 
 DNS-based resources match the address entered and all subdomains.
```
```suggestion
DNS-based resources match the address entered and all subdomains.

e.g. `github.com` will match `imgs.github.com` and `github.com` itself, including all SSH and HTTPS traffic.
```

jamilbk replied:
Updated this section to be much more detailed and provide examples

ReactorScram commented on website/src/components/FAQ/readme.mdx:
```diff
@@ -229,4 +198,7 @@ remains inaccessible from the internet, with no visible entry points.
 
 Firezone helps improve organizations’ cybersecurity posture by offering a modern
 approach to securing access to sensitive Resources in their private network.
```
nit: I get "Gateway" and "Client" if it's short for "Firezone Gateway" and "Firezone Client" but this isn't really a proper noun, right?

```suggestion
approach to securing access to sensitive resources in their private network.
```

jamilbk replied:
Yeah I was trying to determine how to think about that. In Firezone context it's a proper noun, but since here we preface it with "sensitive" then I think that makes it a generic noun.

ReactorScram commented on website/src/components/SupportOptions/index.tsx:
```diff
@@ -5,34 +5,37 @@ import "@docsearch/css";
+        <p>Try asking on one of our community-powered support channels:</p>
+        <ul>
+          <li>
+            <a href="https://discourse.firez.one/?utm_source=docs.firezone.dev">
+              Discussion forums
```
Didn't know we had a Discourse. Should I have an account on there or just assume anything big will bubble up to Github / Slack?

jamilbk replied:
Invite sent

ReactorScram reviewed (changes requested):
I wrote my comments as I was reading, so some earlier comments will ask questions that were answered later on.

There's a few things I think might confuse first-time users, since they confused me a little.

bmanifold commented on website/src/app/kb/deploy/sites/readme.mdx:
```diff
@@ -1,17 +1,39 @@
+
+A site can be as small as a single gateway running on a single server managing
+the access to the resources on that server, or as large as an entire data center
+network managing access to an entire cluster of servers.
```
```suggestion
network managing access to multiple clusters of servers.
```

The use of `entire` twice in one sentence just felt unusual to me.  Just a personal preference.

bmanifold commented on website/src/app/kb/deploy/sites/readme.mdx:
```diff
@@ -1,17 +1,39 @@
+the access to the resources on that server, or as large as an entire data center
+network managing access to an entire cluster of servers.
+
+## Creating a site
```
I'll let this comment stand in for all places in the docs where this is the same...

Should we capitalize things like `Site`, `Resource`, `Gateway`, etc... regardless of where they occur in the sentence, since they refer to specific types within our system?  I feel like I've seen this in other software documentation and just wanted to throw it out as a suggestion since a lot of our terminology within our system is close to generic terminology.

jamilbk replied:
Yeah, I think it makes sense to capitalize them all whenever they're referring to Firezone entities.

For example, "Create a Resource", but not "Firezone secures access to your company's most sensitive resources".

bmanifold replied:
Yeah, that was exactly how I was thinking about it as well.

bmanifold commented on website/src/app/kb/deploy/gateways/readme.mdx:
```diff
@@ -64,32 +106,56 @@ This means you can automate deployment of the gateway using your automation
+  disconnect from that gateway and ask the portal for a new, healthy one to
+  connect to.
+- On average, the client keepalive timer expires after **20 seconds**. This is
+  the maximum time it takes for existing client connections to be rerouted to a
```
I feel like these two sentences may be a little confusing.  At first it's mentioned that 20 seconds is the average and then in the next sentence it is mentioned that it's the max.

bmanifold commented on website/src/app/kb/deploy/gateways/readme.mdx:
```diff
@@ -64,32 +106,56 @@ This means you can automate deployment of the gateway using your automation
+- Clients detect unhealthy gateways by maintaining keepalive timers with
+  gateways they're connected to. If the keepalive timer expires, the client will
+  disconnect from that gateway and ask the portal for a new, healthy one to
+  connect to.
```
```suggestion
- Clients identify unhealthy gateways by using keepalive timers. If the timer expires, the client disconnects from the unhealthy gateway and requests a new, healthy one from the portal.
```

Personal preference suggestion.  Feel free to ignore/disagree.

bmanifold commented on website/src/app/kb/deploy/gateways/readme.mdx:
```diff
@@ -64,32 +106,56 @@ This means you can automate deployment of the gateway using your automation
+When a client wants to connect to a resource, Firezone randomly picks a random
+healthy gateway in the site to serve the resource being requested. The client
+will then stay connected to that gateway until either the client disconnects or
+the gateway becomes unhealthy.
```
```suggestion
When a client wants to connect to a resource, Firezone randomly selects a healthy gateway within the site to handle the request. The client maintains the connection to that gateway until either the client decides to disconnect or the gateway becomes unhealthy.
```

Mostly a personal preference suggestion, other than the change of `...randomly picks a random...`

bmanifold commented on website/src/app/kb/deploy/policies/readme.mdx:
```diff
@@ -1,19 +1,26 @@
+<Alert color="warning">
+  **Note:** To preserve audit trails, policy access cannot be changed once a
+  policy is created. Double-check to ensure the group and resource selected are
+  correct before creating the policy.
```
I know this isn't the intent, but it almost feels like this text is saying, once access is granted it can never be revoked.  Maybe we should add a sentence specifying that policies can be deleted to remove access?

jamilbk committed:
Fix linter

jamilbk committed:
Address @ReactorScram's feedback

bmanifold commented on website/src/app/kb/authenticate/oidc/readme.mdx:
```diff
@@ -1,29 +1,116 @@
 - [Fusion Auth](/kb/authenticate/oidc/fusion)
 
-For others, use the universal setup instructions below.
+Fo others, consult your provider's documentation for setting up an OpenID
```
```suggestion
For all other identity providers, consult your provider's documentation for setting up an OpenID
```

bmanifold commented on website/src/app/kb/authenticate/oidc/readme.mdx:
```diff
@@ -1,29 +1,116 @@
+  Firezone account and are used to complete the authentication process. These
+  are configured in your identity provider's OAuth app settings.
+- [Client ID and secret](#client-id-and-secret): These are used to authenticate
+  Firezone with your identity provider. These are configured in Firezone.
```
```suggestion
  Firezone with your identity provider. These are generated in your identity provider and then configured in Firezone.
```

bmanifold commented on website/src/app/kb/authenticate/oidc/readme.mdx:
```diff
@@ -1,29 +1,116 @@
+  Firezone with your identity provider. These are configured in Firezone.
+- [Discovery document URI](#discovery-document-uri): This is the URL to your
+  identity provider's OIDC discovery document. This is used to automatically
+  configure the connector with your identity provider's settings and is
```
```suggestion
  configure the connector with your identity provider's settings. The URL is generated by your identity provider and then
```

bmanifold commented on website/src/app/kb/authenticate/oidc/readme.mdx:
```diff
@@ -1,29 +1,116 @@
+provider in your Firezone account. They allow Firezone to receive authentication
+tokens from your identity provider to complete the authentication process.
+
+### Client ID and secret
```
```suggestion
### Client ID and client secret
```
This is a nit, so feel free to ignore if you don't like it.

bmanifold commented on website/src/app/kb/authenticate/oidc/readme.mdx:
```diff
@@ -1,29 +1,116 @@
+
+### Client ID and secret
+
+You'll also need to provide the client ID and secret from your identity provider
```
```suggestion
You'll also need to provide the client ID and client secret from your identity provider
```
Again, another nit.

bmanifold commented on website/src/app/kb/authenticate/google/readme.mdx:
```diff
@@ -1,56 +1,51 @@
 Setting up the Google Workspace connector is similar to the process of setting
 up a universal OIDC connector. The main difference is the addition of a few
-extra scopes needed to enable user/group sync.
+extra read-only scopes needed to enable sync.
```
I noticed you replaced `user / group` with `directory` above.  Would it make sense to put it here as well?

bmanifold commented on website/src/app/kb/authenticate/entra/readme.mdx:
```diff
@@ -0,0 +1,424 @@
+  />
+</Link>
+
+Enter **Firezone Connector** for the description, then select an appropriate
+lifetime for the secret and click **Add**.
```
Wondering if we should add a small info section here saying something like "Please be aware that because MS doesn't offer indefinite lifetime secrets, you will be responsible for rotating this secret as needed.  If the token expires, directory sync will stop and all users authenticating with MS Entra will not be able to login...."

bmanifold commented on website/src/app/kb/authenticate/entra/readme.mdx:
```diff
@@ -0,0 +1,424 @@
+</Link>
+
+**Important**: Make sure to save the `Client ID` and `Client secret` fields in a
+safe place as they won't be shown again.
```
```suggestion
safe place as the `Client secret` won't be shown again.
```

bmanifold commented on website/src/app/kb/authenticate/entra/readme.mdx:
```diff
@@ -0,0 +1,424 @@
+If you get successfully redirected back to your Firezone admin dashboard, you're
+done! Your Entra ID provider is now successfully configured. Users in your
+organization can now authenticate to Firezone using this provider and user /
+group sync will occur every few minutes.
```
```suggestion
done! Your Entra ID provider is now successfully configured. Users in your
organization will be able to authenticate to Firezone using this provider after the first directory sync.
Directory sync happens every few minutes and the status is shown on the identity providers page in the portal.
```

bmanifold commented on website/src/app/kb/authenticate/okta/readme.mdx:
```diff
@@ -0,0 +1,262 @@
+The Firezone Okta connector integrates with Okta's APIs to support user
+authentication and directory sync.
+
+Users, groups, and organizational units are synced every few minutes to ensure
```
Was this copied from the Google Workspace page?  I don't think Okta has organizational units does it?

bmanifold commented on website/src/app/kb/authenticate/okta/readme.mdx:
```diff
@@ -0,0 +1,262 @@
+
+Setting up the Okta connector is similar to the process of setting up a
+universal OIDC connector. The main difference is the addition of a few extra
+read-only scopes needed to enable sync.
```
```suggestion
read-only scopes needed to enable directory sync.
```

bmanifold commented on website/src/app/kb/authenticate/okta/readme.mdx:
```diff
@@ -0,0 +1,262 @@
+[Download the Firezone logo](/images/kb/authenticate/okta/firezone-logo.png) to
+use for the app integration and upload it to Okta (optional).
+
+Ensure the **Refresh Token** grant type is selected.
```
```suggestion
Ensure both the **Authorization Code** and **Refresh Token** grant types are selected.
```

bmanifold commented on website/src/app/kb/authenticate/okta/readme.mdx:
```diff
@@ -0,0 +1,262 @@
+  />
+</Link>
+
+In the **Assignments** tab, assign the app to the groups you want to have access
+to Firezone.
```
Was thinking we probably want to put a note underneath this saying something like:
```
"Please note, the user setting up this connector must have access to the Okta application
being created.  If you choose to "skip group assignment for now", please make sure your 
user is assigned to this Okta app before finishing the setup in Firezone..."
```

bmanifold commented on website/src/app/kb/authenticate/okta/readme.mdx:
```diff
@@ -0,0 +1,262 @@
+If you get successfully redirected back to your Firezone admin dashboard, you're
+done! Your Okta provider is now successfully configured. Users in your
+organization can now authenticate to Firezone using this provider and user /
+group sync will occur every few minutes.
```
Same comment here as in the MS Entra ID page, just letting people know that users can login after the first directory sync.

jamilbk committed:
checkpoint capitalization

jamilbk committed:
Finish addressing @bmanifold's feedback

jamilbk committed:
Merge branch 'main' into docs/1.0.0-pre.9

ReactorScram commented on website/src/app/kb/deploy/clients/readme.mdx:
```diff
@@ -1,19 +1,57 @@
+
+## Provision with MDM
+
+Provisioning the Firezone client onto end-user devices should work out of the
```
```suggestion
Provisioning the Firezone Client onto end-user devices should work out of the
```

?

ReactorScram commented on website/src/app/kb/deploy/dns/readme.mdx:
```diff
@@ -0,0 +1,89 @@
+
+## How DNS works in Firezone
+
+Each Firezone client embeds a lightweight DNS proxy used to route queries to an
```
Capitalize Client here or no?

ReactorScram commented on website/src/app/kb/deploy/gateways/readme.mdx:
```diff
@@ -31,65 +37,124 @@ you'll need to make sure the following outbound traffic is allowed:
+-- in some cases, even on the same physical host. This ensures the lowest
+possible latency and highest possible throughput for client connections, and
+allows you to more easily deploy additional Gateways for a particular Resource
+that needs to handle more client connections.
```
```suggestion
possible latency and highest possible throughput for Client connections, and
allows you to more easily deploy additional Gateways for a particular Resource
that needs to handle more Client connections.
```

Can probably capitalize here since The Firezone Client is the only thing that will connect to The Firezone Gateway

ReactorScram replied:
Well the diff would be huge 🤷‍♀️ either way is fine

ReactorScram commented on website/src/app/kb/deploy/gateways/readme.mdx:
```diff
@@ -31,65 +37,124 @@ you'll need to make sure the following outbound traffic is allowed:
-most gateways should be able to saturate 1 Gbps with relative ease. Faster
-speeds can be achieved with syscall optimization and multi-threaded
-optimizations, both of which are currently being worked on.
+A single client connection to a single Gateway can typically reach speeds of **1
```
```suggestion
A single Client connection to a single Gateway can typically reach speeds of **1
```

ReactorScram commented on website/src/app/kb/deploy/gateways/readme.mdx:
```diff
@@ -31,65 +37,124 @@ you'll need to make sure the following outbound traffic is allowed:
-speeds can be achieved with syscall optimization and multi-threaded
-optimizations, both of which are currently being worked on.
+A single client connection to a single Gateway can typically reach speeds of **1
+Gbps** or more. This scales linearly with each client connection up to the
```
```suggestion
Gbps** or more. This scales linearly with each Client connection up to the
```

ReactorScram commented on website/src/app/kb/deploy/gateways/readme.mdx:
```diff
@@ -31,65 +37,124 @@ you'll need to make sure the following outbound traffic is allowed:
+  Gateway in the event of a Gateway failure.
+
+By using two independent health checks in the portal and the Client, Firezone
+ensures that temporary network issues between the Client and portal do not
```
```suggestion
ensures that temporary network issues between the Client and Portal do not
```

ReactorScram commented on website/src/app/kb/deploy/gateways/readme.mdx:
```diff
@@ -31,65 +37,124 @@ you'll need to make sure the following outbound traffic is allowed:
-Two or more Gateways deployed in the same site will automatically fail over and
-load balance for each other.
+Similar to automatic failover, two or more Gateways deployed within a Site will
+automatic load balance client connections to Resources in that Site.
```
```suggestion
automatically load balance client connections to Resources in that Site.
```

ReactorScram commented on website/src/app/kb/deploy/policies/readme.mdx:
```diff
@@ -1,19 +1,26 @@
-Policies are what grant users access to resources.
+</PlanBadge>
+
+Policies are what grant users access to Resources.
```
```suggestion
Policies grant users access to Resources.
```

Nit: A little more active voice with one less pronoun if you need it

ReactorScram commented on website/src/app/kb/deploy/policies/readme.mdx:
```diff
@@ -1,19 +1,26 @@
-To define a policy, go to `Policies` -> `Add Policy`.
+<Alert color="warning">
+  **Note:** To preserve audit trails, Policy access cannot be changed once a
+  Policy is created. Double-check to ensure the group and resource selected are
```
```suggestion
  Policy is created. Double-check to ensure the group and Resource selected are
```

ReactorScram commented on website/src/app/kb/deploy/resources/readme.mdx:
```diff
@@ -1,20 +1,48 @@
+- **CIDR**: A range of IPv4 or IPv6 addresses in CIDR notation, such as
+  `10.1.2.0/24` or `2001:db8::/48`
+
+Firezone does not currently filter traffic based on any layer 4 details such as
+transport protocol (UDP / TCP) or port number. If you're interested in a feature
```
Good to know, I didn't notice when it was removed 👍

ReactorScram commented on website/src/app/kb/deploy/sites/readme.mdx:
```diff
@@ -1,17 +1,39 @@
+</PlanBadge>
+
+Sites represent a shared network environment that Gateways and Resources exist
+within. All Gateways and Resources in a Site must have **unobstructed network
+connectivity** to each other.
```
```suggestion
within. All Gateways in a Site must have **unobstructed network
connectivity** to all Resources in the Site.
```

The Gateways never connect to other Gateways and we don't need Resources to connect to each other since our scope is only that Gateways connect to Resources, right? Resources don't even need to be able to connect to Gateways if there's a trapdoor firewall for some reason.

ReactorScram commented on website/src/app/kb/deploy/sites/readme.mdx:
```diff
@@ -1,17 +1,39 @@
+
+A Site can be as small as a single Gateway running on a single server managing
+the access to the Resources on that server, or as large as a data center network
+managing access to an cluster of servers.
```
```suggestion
managing access to a cluster of servers.
```

ReactorScram commented on website/src/app/kb/quickstart/readme.mdx:
```diff
@@ -13,35 +23,38 @@ with the some of the most common configuration options.
-- Server or VM (with the ability to deploy a docker container or Linux
-  application) that can connect to both the resource, and the internet
+  SaaS application, or subnet)
+- Server or VM you're able to deploy a Docker container or Linux binary on that
```
nit

```suggestion
- x86_64 Server or VM you're able to deploy a Docker container or Linux binary on that
```

ReactorScram commented on website/src/app/kb/quickstart/readme.mdx:
```diff
@@ -13,35 +23,38 @@ with the some of the most common configuration options.
 
 ## Summary
 
 1. **Sign in to your Firezone Admin Portal** (e.g.
```
"Admin Portal" is capitalized here because it's title case, and not because "Admin Portal" is generally capitalized, right?

ReactorScram commented on website/src/app/kb/quickstart/readme.mdx:
```diff
@@ -50,69 +63,83 @@ guide.
 
-Clicking "Deploy a Gateway" on the Site Details page will give you two options
+Clicking **Deploy a Gateway** on the Site Details page will give you two options
 for deploying a Gateway: `Docker` or `Systemd` (choose whichever method you
```
```suggestion
for deploying a Gateway: `Docker` or `systemd` (choose whichever method you
```

They actually seem to stylize 'systemd' in lower case https://systemd.io/

ReactorScram commented on website/src/app/kb/quickstart/readme.mdx:
```diff
@@ -50,69 +63,83 @@ guide.
+- CIDR Range (e.g. `10.10.10.0/24` or `2001:db8::/48`)
+
+Optionally, you're given the option to provide an `Address description` which,
+if provided, will be what's shown as the Resource address in the clients when
```
```suggestion
if provided, will be what's shown as the Resource address in the Clients when
```

ReactorScram commented on website/src/app/kb/reference/glossary/readme.mdx:
```diff
@@ -0,0 +1,50 @@
+to/from Resources associated with a Site will pass through one of that Site’s
+Gateways. We distribute the Gateway as self-contained binaries on our
+[releases page](https://github.com/firezone/firezone/releases), or as a Docker
+image or Systemd unit file with instructions shown when you deploy the Gateway
```
```suggestion
image or systemd unit file with instructions shown when you deploy the Gateway
```

ReactorScram commented on website/src/app/kb/reference/glossary/readme.mdx:
```diff
@@ -0,0 +1,50 @@
+image or Systemd unit file with instructions shown when you deploy the Gateway
+from the admin portal. Gateways can run on everything from a Raspberry Pi to
+bare metal servers and everything in between. Gateways are designed to be
+lightweight and don't require persistent storage to function.
```
So even if you image 100 IoT devices with the same service token, they can all run Gateways? Or do they need unique IDs?

AIUI the Gateway doesn't need to write anything to disk but does it need any unique read-only data?

ReactorScram commented on website/src/app/kb/reference/glossary/readme.mdx:
```diff
@@ -0,0 +1,50 @@
+**Group**: [User groups](/kb/deploy/groups) consist of one or more users,
+usually members of the same team or department (e.g. Engineering, DevOps, Sales)
+and can be used in Policies to give all users in that Group access to a
+Resource. Users and groups can be automatically synced from your Identity
```
```suggestion
Resource. Users and Groups can be automatically synced from your Identity
```

ReactorScram commented on website/src/app/kb/reference/glossary/readme.mdx:
```diff
@@ -0,0 +1,50 @@
+access to your Resources.
+
+**Policy**: [Policies](/kb/deploy/policies) define a one-to-one mapping between
+a user group and a Resource. Access to Resources is default-deny, which means a
```
```suggestion
a user Group and a Resource. Access to Resources is default-deny, which means a
```

ReactorScram commented on website/src/app/kb/reference/glossary/readme.mdx:
```diff
@@ -0,0 +1,50 @@
+
+**Site**: [Sites](/kb/deploy/sites) are user-created environments where admins
+can manage Resources and the Gateways that enable access to those Resources
+(e.g. US-West, Chicago-LAN, or Prod). All Gateways and Resources in a Site are
+assumed to be able to reach each other in a shared network context such as a VPC
+or LAN.
```
Same question as above, Gateways don't actually need to be able to reach each other, the only requirement is that all Gateways can reach all Resources, even if Resources can't make any egress connections and Gateways somehow can't see each other, right?

ReactorScram commented on website/src/components/FAQ/readme.mdx:
```diff
@@ -144,15 +113,15 @@ failover.
 
 Service accounts facilitate secure connections from a server, VM or container to
 a Resource. Because service accounts do not involve a user, they can’t fulfill
-2FA requirements, and use the Firezone client in a non-interactive way (headless
+2FA requirements, and use the Firezone Client in a non-interactive way (headless
 mode).
```
Github isn't letting me add a suggestion here but

This could parse as "Because services accounts do not involve a user, they can't (fulfill requirements, and use the Firezone Client in a non-interactive way)"

Consider "Because service accounts do not involve a user, they can't (fulfill 2FA requirements), so they (use the Firezone Client in a non-interactive way)"

Or to put the positive first and details second, "Service accounts use the Firezone Client in non-interactive mode ("headless" mode), since they do not involve user interaction and cannot fulfill 2FA requirements."

ReactorScram commented on website/src/components/FAQ/readme.mdx:
```diff
@@ -144,15 +113,15 @@ failover.
 ## Performance
 
 #### Does all my traffic go through Firezone?
 
```
```suggestion
When Clients connect to anything outside your Firezone network, they do not go through Firezone. They are resolved with the upstream DNS or system DNS, and the OS sends traffic over the other network interfaces, ignoring Firezone.
```

ReactorScram commented on website/src/components/FAQ/readme.mdx:
```diff
@@ -144,15 +113,15 @@ failover.
 
 Network traffic is always end-to-end encrypted, and by default, routes directly
-to gateways running on your infrastructure. In rare circumstances, encrypted
+to Gateways running on your infrastructure. In rare circumstances, encrypted
 traffic can pass through our global relay network if a direct connection cannot
 be established. Firezone can never decrypt the contents of your traffic.
```
```suggestion
to Gateways running on your infrastructure. If we can't establish a direct connection,
encrypted traffic may pass through our global relay network. 
Firezone can never decrypt the contents of your traffic.
```

(Unless we get an NSA letter ofc)

ReactorScram commented on website/src/components/FAQ/readme.mdx:
```diff
@@ -182,12 +151,12 @@ approximately 5 minutes before being disconnected.
+#### Do users need to interact with the Firezone Client?
 
 The client app is designed to be unobtrusive, and once a user clicks connect and
 authenticates with their credentials, it should run in the background without
```
```suggestion
The client app is designed to be unobtrusive, and once a user signs in with their credentials, it should run in the background without
```

There isn't a "Connect" button, is there? On Windows it says "Sign In"

ReactorScram reviewed (approved):
🚀  LGTM for Clients and stuff I've used. I glanced over the group sync / SSO stuff but didn't really read it.

I made a lot of comments for capitalization and small grammar changes which are all take-it-or-leave-it nits. e.g. `systemd` is typically written lowercase, not sure what the rule is on `client connections` vs `Client connections`

jamilbk committed:
Checkpoint

jamilbk committed:
Linux docs

ReactorScram commented on website/src/app/kb/user-guides/linux-client/readme.mdx:
````diff
@@ -1,21 +1,56 @@
+```text
+> sudo ./linux-client-x64 -h
+
+Usage: linux-client-arm64 [OPTIONS] --firezone-id <FIREZONE_ID> <TOKEN> [MAX_PARTITION_TIME]
````
hm is it arm64 or x64 here

jamilbk replied:
👍

jamilbk committed:
Address @ReactorScram's feedback
